### PR TITLE
feat: onramp regional ip block (FE integration)

### PR DESCRIPTION
**Pull request by ChefBingbong**
---

## PR-Codex overview
This PR focuses on improving the BuyCrypto feature. 

### Detailed summary
- Added `ProviderAvailabilities` type to `types.ts`.
- Added `parseDisabledProviders` function to `constants.ts`.
- Updated `fetchMercuryoSignedUrl` function in `useIframeUrlFetcher.ts`.
- Removed `setUsersIpAddress` action in `actions.ts`.
- Updated `BuyCryptoPage` component in `index.tsx`.
- Updated `GetQuotesButton` component in `GetQuotesButton.tsx`.
- Removed `userIpAddress` from `BuyCryptoState` in `reducer.ts`.
- Updated `ProviderCampaign` component in `ProviderCampaign.tsx`.
- Updated `fetchProviderAvailabilities` function in `useProviderAvailability.ts`.
- Updated `CryptoQuoteForm` component in `CryptoQuoteForm.tsx`.
- Updated `Accordion` component in `Accordion.tsx`.
- Updated `BuyCrypto` component in `index.tsx`.
- Updated `useBuyCryptoActionHandlers` and `usePriceQuotes` hooks in `hooks.ts`.

> The following files were skipped due to too many changes: `apps/web/src/state/buyCrypto/hooks.ts`, `apps/web/src/views/BuyCrypto/containers/BuyCryptoForm.tsx`, `apps/web/src/views/BuyCrypto/hooks/usePriceQuoter.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

---

### Focus of the PR:
The focus of this PR is to remove the check for disabled providers in the `sortProviderQuotes` function in the `usePriceQuoter` hook.

### Detailed summary:
- Removed the check for disabled providers in the `sortProviderQuotes` function.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by chef-jojo** [comment]
Any reason not using middleware?

**Comment by ChefBingbong** [comment]
> Any reason not using middleware?

do u mean for getting the ip?

File: apps/web/src/pages/api/onramp/provider-availability.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ONRAMP_API_BASE_URL } from 'config/constants/endpoints'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    const xForwardedFor = req.headers['x-forwarded-for']
+    const userIp = xForwardedFor ? (xForwardedFor as string).split(',')[0].trim() : null
+
+    const response = await fetch(`${ONRAMP_API_BASE_URL}/fetch-provider-availability-get?userIp=${userIp}`, {
```
Comment (chefjackson) [inline apps/web/src/pages/api/onramp/provider-availability.ts:9]: why are we adding this request proxy instead of doing the check on the service end?

Comment (chef-jojo) [reply apps/web/src/pages/api/onramp/provider-availability.ts:9]: The service should be able to parse the request header, we don't need this api totally

File: apps/web/src/views/BuyCrypto/hooks/useProviderAvailability.ts [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    })
+    const result = await response.json()
+    return result.result
+  } catch (error) {
+    console.error(error)
+    return { MoonPay: true, Mercuryo: true, Transak: true }
```
Comment (chefjackson) [inline apps/web/src/views/BuyCrypto/hooks/useProviderAvailability.ts:16]: Should be default false?

File: apps/web/src/views/BuyCrypto/hooks/usePriceQuoter.ts [hunk 3]
```diff
@@ -14,34 +15,36 @@ const usePriceQuotes = () => {
...
-      }
+  const { data: providerAvailabilities, isLoading: isAvailabilitiesLoading } = useQuery(
+    ['providerAvailabilities'],
+    () => fetchProviderAvailabilities(),
+    {
+      initialData: {
+        MoonPay: true,
+        Mercuryo: true,
+        Transak: true,
+      },
```
Comment (chefjackson) [inline apps/web/src/views/BuyCrypto/hooks/usePriceQuoter.ts:28]: If we haven't got the availabilities from the endpoint should default false?

File: apps/web/src/pages/api/onramp/provider-availability.ts (hunk 1 above)
Comment (ChefBingbong) [reply apps/web/src/pages/api/onramp/provider-availability.ts]: this has been resolved. added ip to the buy crypto api

**Pull request closed without merging**